### PR TITLE
Fixing #670: restoring allowSnapshots which was disabled in PR #665

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Fixing the problem related to snapshots in UpdateParentMojo + adding a unit test testing snapshots.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@slawekjaranowski please review

**Comment by pzygielo** [comment]
Just linking
- #670

**Comment by pzygielo** [comment]
Still getting
```
[INFO] --- versions-maven-plugin:2.12.1-SNAPSHOT:update-parent (default-cli) @ parent ---
[WARNING] Not updating version: could not resolve any versions
```

**Comment by ajarmoniuk** [comment]
> 

with what invocation arguments? did you use -DallowSnapshots ?

**Comment by pzygielo** [comment]
> with what invocation arguments?
```
mvn org.codehaus.mojo:versions-maven-plugin:2.12.1-SNAPSHOT:update-parent -DparentVersion=71-SNAPSHOT -DallowSnapshots=true
```

**Comment by pzygielo** [comment]
As there is no reproducer in original issue, here's my pom:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <parent>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>mojo-parent</artifactId>
        <version>70</version>
    </parent>

    <groupId>mojohaus</groupId>
    <artifactId>issue-670</artifactId>
    <version>0.31-SNAPSHOT</version>
    <packaging>pom</packaging>

</project>
```

**Comment by ajarmoniuk** [comment]
> ```
>         <groupId>org.codehaus.mojo</groupId>
>         <artifactId>mojo-parent</artifactId>
>         <version>70</version>
> ```

And what are the available versions of that artifact?

**Comment by ajarmoniuk** [comment]
Or perhaps could you please refer to the artifacts in it-repo?

File: src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java [hunk 1]
```diff
@@ -134,15 +134,13 @@ protected void update( ModifiedPomXMLEventReader pom )
         ArtifactVersion artifactVersion;
         try
         {
-            artifactVersion = findLatestVersion( artifact, versionRange, false, true,
-                    allowDowngrade );
+            artifactVersion = findLatestVersion( artifact, versionRange, null, false, allowDowngrade );
```
Comment (ajarmoniuk) [inline src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java:137]: this is going back to the original argument list from before the offending PR; that `null` argument means overriding the property accessible to `findLatestVersions` -- I think this is to be refactored, but that's for another time

The other change is setting `usePluginRepositories` to `false` like it was before that PR

**Comment by ajarmoniuk** [comment]
Thanks for the test case. Amended the PR.

**Comment by ajarmoniuk** [comment]
Also added that test as an it.

**Comment by slawekjaranowski** [comment]
@ajarmoniuk thanks for fix
@pzygielo thanks for testing

**Pull request merged**